hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/InvokeOnMembers.java

```diff
@@ -0,0 +1,135 @@
+import com.hazelcast.spi.OperationService;
+import com.hazelcast.spi.annotation.PrivateApi;
+import com.hazelcast.spi.serialization.SerializationService;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+/**
+ * Executes an operation on a set of targets.
+ * Similar to {@link com.hazelcast.spi.impl.operationservice.impl.InvokeOnPartitions} but for members.
+ */
+@PrivateApi
+public final class InvokeOnMembers {
+
+    private static final int TRY_COUNT = 10;
+    private static final int TRY_PAUSE_MILLIS = 300;
+
+    private final ILogger logger;
+
+    private final OperationService operationService;
+    private final SerializationService serializationService;
+
+    private final String serviceName;
+    private final OperationFactory operationFactory;
+    private final Collection<Member> targets;
+    private final Map<Member, Future> futures;
+    private final Map<Member, Object> results;
+
+    public InvokeOnMembers(NodeEngine nodeEngine,
+                           String serviceName, OperationFactory operationFactory,
+                           Collection<Member> targets) {
+        this.logger = nodeEngine.getLogger(getClass());
+        this.operationService = nodeEngine.getOperationService();
+        this.serializationService = nodeEngine.getSerializationService();
+        this.serviceName = serviceName;
+        this.operationFactory = operationFactory;
+        this.targets = targets;
+        this.futures = new HashMap<Member, Future>(targets.size());
+        this.results = new HashMap<Member, Object>(targets.size());
+    }
+
+    /**
+     * Executes all the operations on the partitions.
+     */
+    public Map<Member, Object> invoke() throws Exception {
+        invokeOnAllTargets();
+
+        awaitCompletion();
+
+        retryFailedTargets();
+
+        return results;
+    }
+
+    private void invokeOnAllTargets() {
+        for (Member target : targets) {
+            Future future = operationService
+                    .createInvocationBuilder(serviceName, operationFactory.createOperation(), target.getAddress())
+                    .setTryCount(TRY_COUNT)
+                    .setTryPauseMillis(TRY_PAUSE_MILLIS)
+                    .invoke();
+            futures.put(target, future);
+        }
+    }
+
+    private void awaitCompletion() {
+        for (Map.Entry<Member, Future> responseEntry : futures.entrySet()) {
+            try {
+                Future future = responseEntry.getValue();
+                results.put(responseEntry.getKey(), serializationService.toObject(future.get()));
+            } catch (Throwable t) {
+                if (logger.isFinestEnabled()) {
+                    logger.finest(t);
+                } else {
+                    logger.warning(t.getMessage());
+                }
```

Comment:
Kept it consistent with the partitions version, I also thought was a bit convoluted. I can look at this again once we merge the logic, wdyt ?